tree/BinaryTree.js

```diff
@@ -0,0 +1,51 @@
+
+
+class BinaryTree {
+  constructor(root = null){
+    this.root = root;
+    this.storage = [];
+  }
+
+
+  preOrder(node = this.root){
+    this.logNodePre(node);
+    this.storage.unshift(this.root);
```

Comment:
To follow the constraints of traversing a tree, you would use push in this step, not unshift. If you use unshift, you'll track the nodes in opposite order.